crates/bevy_ecs/src/system/exclusive_function_system.rs

```diff
@@ -0,0 +1,228 @@
+    query::Access,
+    schedule::{SystemLabel, SystemLabelId},
+    system::{
+        check_system_change_tick, AsSystemLabel, ExclusiveSystemParam, ExclusiveSystemParamFetch,
+        ExclusiveSystemParamItem, ExclusiveSystemParamState, IntoSystem, System, SystemMeta,
+        SystemTypeIdLabel,
+    },
+    world::{World, WorldId},
+};
+use bevy_ecs_macros::all_tuples;
+use std::{borrow::Cow, marker::PhantomData};
+
+/// A function system that runs with exclusive [`World`] access.
+///
+/// You get this by calling [`IntoSystem::into_system`]  on a function that only accepts
+/// [`ExclusiveSystemParam`]s. The output of the system becomes the functions return type, while the input
+/// becomes the functions [`In`] tagged parameter or `()` if no such parameter exists.
+///
+/// [`ExclusiveFunctionSystem`] must be `.initialized` before they can be run.
+pub struct ExclusiveFunctionSystem<In, Out, Param, Marker, F>
+where
+    Param: ExclusiveSystemParam,
+{
+    func: F,
+    param_state: Option<Param::Fetch>,
+    system_meta: SystemMeta,
+    world_id: Option<WorldId>,
+    archetype_generation: ArchetypeGeneration,
+    // NOTE: PhantomData<fn()-> T> gives this safe Send/Sync impls
+    marker: PhantomData<fn() -> (In, Out, Marker)>,
+}
+
+pub struct IsExclusiveFunctionSystem;
+
+impl<In, Out, Param, Marker, F> IntoSystem<In, Out, (IsExclusiveFunctionSystem, Param, Marker)>
+    for F
+where
+    In: 'static,
+    Out: 'static,
+    Param: ExclusiveSystemParam + 'static,
+    Marker: 'static,
+    F: ExclusiveSystemParamFunction<In, Out, Param, Marker> + Send + Sync + 'static,
+{
+    type System = ExclusiveFunctionSystem<In, Out, Param, Marker, F>;
+    fn into_system(func: Self) -> Self::System {
+        ExclusiveFunctionSystem {
+            func,
+            param_state: None,
+            system_meta: SystemMeta::new::<F>(),
+            world_id: None,
+            archetype_generation: ArchetypeGeneration::initial(),
+            marker: PhantomData,
+        }
+    }
+}
+
+const PARAM_MESSAGE: &str = "System's param_state was not found. Did you forget to initialize this system before running it?";
+
+impl<In, Out, Param, Marker, F> System for ExclusiveFunctionSystem<In, Out, Param, Marker, F>
+where
+    In: 'static,
+    Out: 'static,
+    Param: ExclusiveSystemParam + 'static,
+    Marker: 'static,
+    F: ExclusiveSystemParamFunction<In, Out, Param, Marker> + Send + Sync + 'static,
+{
+    type In = In;
+    type Out = Out;
+
+    #[inline]
+    fn name(&self) -> Cow<'static, str> {
+        self.system_meta.name.clone()
+    }
+
+    #[inline]
+    fn component_access(&self) -> &Access<ComponentId> {
+        self.system_meta.component_access_set.combined_access()
+    }
+
+    #[inline]
+    fn archetype_component_access(&self) -> &Access<ArchetypeComponentId> {
+        &self.system_meta.archetype_component_access
+    }
+
+    #[inline]
+    fn is_send(&self) -> bool {
+        self.system_meta.is_send()
+    }
```

Comment:
Should this just be hardcoded to `true`?

Reply:
We could toss in a debug assert here IMO to verify that the data matches.

Reply:
Hmm being NonSend isn't _required_ for exclusive system execution, although in practice that might be useful (to ensure that exclusive systems can always access NonSend resources) and it would come for free (because exclusive systems will run on the main thread, at least for now). In fact, it shouldn't matter what we set this to (for now). I'm cool with encoding this behavior as a hard requirement for now, rather than something implicit to the scheduler.

We might want to walk that back later though if we choose to expand exclusive system execution options.

Reply:
Done!